Lib/_pyio.py

```diff
@@ -2529,3 +2530,23 @@ def encoding(self):
     def detach(self):
         # This doesn't make sense on StringIO.
         self._unsupported("detach")
+
+
+# ____________________________________________________________
+
+import atexit, weakref
+
+_all_writers = weakref.WeakSet()
+
+def _register_writer(w):
+    # keep weak-ref to buffered writer
+    _all_writers.add(w)
+
+def _flush_all_writers():
+    # Ensure all buffered writers are flushed before proceeding with
+    # normal shutdown.  Otherwise, if the underlying file objects get
+    # finalized before the buffered writer wrapping it then any buffered
+    # data will be lost.
+    for w in _all_writers:
+        w.flush()
```

Comment:
I think that would be wise.  flush() probably should not raise an error but if it does then not catching it would be bad.  Do you think just try/except (base) and just swallow it?  I think that would be okay because we really don't expect flush() to raise an error and if it does, there is nothing much to do about it.